src/plopp/graphics/lineview.py

```diff
@@ -102,64 +103,25 @@ def __init__(
             **kwargs,
         )
         self.canvas.yscale = norm
-
-        # The line view is potentially plotting many lines. When ``render`` is called,
-        # it calls ``update`` for each line. At the end of ``update``, a range autoscale
-        # is applied. We want to avoid autoscaling at the end of each ``update`` call,
-        # because searching for limits could be expensive.
-        self._no_autoscale = True
         self.render()
-        self._no_autoscale = False
-        self.canvas.autoscale()
         self.canvas.finalize()
 
-    def update(self, new_values: sc.DataArray, key: str):
-        """
-        Add new line or update line values.
-
-        Parameters
-        ----------
-        new_values:
-            New data to create or update a :class:`Line` object from.
-        key:
-            The id of the node that sent the new data.
-        """
-        if new_values.ndim != 1:
-            raise ValueError("LineView can only be used to plot 1-D data.")
```

Comment:
Maybe I missed something but it seems we loose this error message here. Is there another check elsewhere using the new `_ndim` attribute?

Reply:
Yes, at the start of the `update` function in the `GraphicalView` ;-)